lib/ui/bottom-panel.js

```diff
@@ -8,65 +8,98 @@ export class BottomPanel {
   constructor(scope) {
     this.subscriptions = new CompositeDisposable
     this.element = document.createElement('linter-panel') // TODO(steelbrain): Make this a `div`
-    this.element.tabIndex = "-1"
+    this.element.tabIndex = '-1'
+    this.messagesElement = document.createElement('div')
     this.panel = atom.workspace.addBottomPanel({item: this.element, visible: false, priority: 500})
     this.visibility = false
+    this.visibleMessages = 0
     this.scope = scope
     this.messages = new Map()
 
+    // Keep messages contained to measure height.
+    this.element.appendChild(this.messagesElement)
+
+    this.subscriptions.add(atom.config.observe('linter.alwaysTakeMinimumSpace', config => {
+      this.takeMinimumHeight = config
+      this.updateHeight()
+    }))
+
+    this.subscriptions.add(atom.config.observe('linter.errorPanelHeight', height => {
```

Comment:
No, they were all already unnecessary actually. You only need the one `updateVisibility` and I've fixed that now. And actually, I don't think you need that since `setMessages` calls `updateVisibility` which is when you need it.